src/networking.c

```diff
@@ -1579,6 +1580,10 @@ int writeToClient(client *c, int handler_installed) {
             return C_ERR;
         }
     }
+
+    if (c->obuf_soft_limit_reached_time && !checkClientOutputBufferLimits(c))
+        queueClientForReprocessing(c);
```

Comment:
This isn't thread safe, probably need to do this after threading.

Reply:
thanks. so in `sendReplyToClient`, `handleClientsWithPendingWrites`, and `handleClientsWithPendingWritesUsingThreads` should be good?

Reply:
Yeah, that seems correct